app/src/androidTest/java/com/github/orkest/View/sharedMusic/SharedMusicPostUITest.kt

```diff
@@ -0,0 +1,91 @@
+package com.github.orkest.View.sharedMusic
+
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.orkest.Constants
+import com.github.orkest.Model.Profile
+import com.github.orkest.Model.Song
+import com.github.orkest.View.sharing.ui.theme.OrkestTheme
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class SharedMusicPostUITest {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Before
+    fun setUp(){
+        val song = Constants.DUMMY_RUDE_BOY_SONG
+        val message = "It's amazing."
+        val profile = Profile(username = "JohnDoe")
+
+        // Launch the composable function
+        composeTestRule.setContent {
+            OrkestTheme {
+                sharedMusicPost(profile, song, message)
+            }
+        }
+    }
+
+
+    @Test
+    fun songCoverIsDisplayed(){
+        composeTestRule.onNodeWithContentDescription("Cover of the album")
+            .assertIsDisplayed()
+    }
+
+    // Test if the song name, author, and album are displayed correctly
+    @Test
+    fun songInfoAreDisplayed(){
+        composeTestRule.onNodeWithText("Rude Boy")
+            .assertIsDisplayed()
+        composeTestRule.onNodeWithText("Rihanna (Rated R)")
+            .assertIsDisplayed()
+    }
+
+    @Test
+    fun profilePictureIsDisplayed(){

```

Comment:
It would also make sense to test if the profile picture (and any other media fetched from the database) is correctly loaded before being displayed. We can maybe add these tests once the backend is implemented